pre_commit_hooks/requirements_txt_fixer.py

```diff
@@ -1,31 +1,41 @@
 from __future__ import print_function
 
 import argparse
-from typing import IO
+import re
+from typing import Dict
 from typing import List
 from typing import Optional
 from typing import Sequence
 
+from pip._internal.download import PipSession
+from pip._internal.req import InstallRequirement
+from pip._internal.req.req_file import parse_requirements
```

Comment:
we can't import private implementation detail from `pip` -- this won't work generically and won't continue to work in the future

Reply:
I do agree with that.
If we don't use this, we'll have to handle each possible variation of requirements file.
IMO using this is better than reinventing the wheel.

Reply:
I think it's fine to only support a subset of requirement types (as we're already doing) and add particular patterns as they are encountered

Reply:
I've made the required changes.